core/src/main/java/org/apache/calcite/runtime/SqlFunctions.java

```diff
@@ -3797,6 +3798,16 @@ public static List distinct(List list) {
     return new ArrayList<>(result);
   }
 
+  /** Support the MAP_KEYS function. */
+  public static List mapKeys(Map map) {
+    return (List) map.keySet().stream().collect(Collectors.toList());
```

Comment:
Then we should handle null here? It seems a NPE Exception will be thrown?

Reply:
@zoudan not needs. the runtime will does it like other engine
![Uploading image.png…]()

Reply:
and why i doesn't add unit test map_keys(cast(null as map[keyType, valueType])) is because the calcite doesn't supported, which you can see here https://github.com/apache/calcite/pull/3189
java.lang.RuntimeException: Error while parsing query: select cast(map['washington', 1, null, 44] as map[CHAR(10), INTEGER NOT NULL>] from (values (1))
	at org.apache.calcite.sql.test.AbstractSqlTester.parseAndValidate(AbstractSqlTester.java:157)
	at org.apache.calcite.sql.test.AbstractSqlTester.validateAndThen(AbstractSqlTester.java:248)
	at org.apache.calcite.sql.test.AbstractSqlTester.checkColumnType(AbstractSqlTester.java:177)
	at org.apache.calcite.test.SqlValidatorFixture.columnType(

Reply:
Why not use `new ArrayList<>(map.keySets())`?